src/Okta.Sdk.IntegrationTests/CollectionScenarios.cs

```diff
@@ -56,6 +58,121 @@ public async Task EnumerateCollectionWithLinq()
 
                 await Task.WhenAll(tasks);
             }
+
+            successful.Should().BeTrue();
+        }
+
+        /// <summary>
+        /// Tests that a developer can use the PagedCollectionEnumerator to enumerate a collection.
+        /// In most cases it is easier to use LINQ, but this approach is available if you access to each page.
+        /// </summary>
+        /// <returns>The asynchronous test.</returns>
+        [Fact]
+        public async Task EnumerateCollectionManually()
+        {
+            var successful = true;
+            var client = TestClient.Create();
+            var createdUsers = new ConcurrentBag<IUser>();
+
+            try
+            {
+                // Create 10 users (in parallel)
+                var tasks = new List<Task>();
+                for (var i = 0; i < 10; i++)
+                {
+                    tasks.Add(CreateRandomUser(client, createdUsers));
+                }
+
+                await Task.WhenAll(tasks);
+
+                // Alright, all set up. Try enumerating users by pages of 2:
+                var retrievedUsers = new List<IUser>();
+                var users = client.Users.ListUsers(limit: 2, filter: "profile.lastName eq \"CollectionEnumeration\"");
+                var enumerator = users.GetPagedEnumerator();
+
+                while (await enumerator.MoveNextAsync())
+                {
+                    retrievedUsers.AddRange(enumerator.CurrentPage.Items);
+                }
+
+                retrievedUsers.Count.Should().Be(10);
+            }
+            catch (Exception e)
+            {
```

Comment:
Don't you have the exception when it fails already?

Reply:
Without this `catch`, the test will appear to pass even if an error occurs, because the `finally` block will swallow the exception and complete successfully. This `catch` and `successful.Should().BeTrue()` at the end ensures the test shows up as a failure if something goes wrong.